### DISCUSSION TITLE
smartdns上游配置119.29.29.29时，ipv6地址返回的时间特别长。

### DISCUSSION THREAD

**Discussion by yyysuo** [pymumu/smartdns#1365] [General]
测速模式和响应模式都是默认，ipv6地址返回的时间普遍在几百ms，打开网页也感觉到慢，比如京东的首页，图片加载的速度明显比配置223.5.5.5差一些。大部分域名其实是没有ipv6地址的，你们有没有相同的现象？

**Comment by yyysuo** [comment]
用223.5.5.5就很快。

**Comment by PikuZheng** [comment]
阿里快的原因是因为它大缓存，在有缓存（即使已经缓存过期）的情况下直接从缓存返回所以快（约等于smartdns的serve-expired功能）。dnspod是要一直递归查询到SOA，所以比阿里慢。
对于一个域名的首次查询（阿里没有缓存过的情况下），我记得曾测试过，在我这里dnspod（走腾讯云北京的独立ip）略快。参考#1233

至于119.29.29.29和223.5.5.5，我认为不排除运营商污染。119.29.29.29不支持https或tls或别的什么可靠连接。

**Comment by pymumu** [comment]
开启cache没有？缓存大小是多少

**Reply by yyysuo** [reply]
> 开启cache没有？缓存大小是多少

开了，开机很长一段时间内，会有很多域名的AAAA记录响应时间在200ms以上，只有119有此表现。配置如下：

server-name smartdns
dualstack-ip-selection no
prefetch-domain yes
serve-expired yes
cache-size 100000000
dnsmasq-lease-file /tmp/dhcp.leases
rr-ttl-min 1
log-size 64K
log-num 1
log-level error
cache-persist yes
cache-file /etc/smartdns/smartdns.cache
resolv-file /tmp/resolv.conf.d/resolv.conf.auto
bind [::]:6053 
bind-tcp [::]:6053 
bind [::]:5555  -no-speed-check -group foreign -no-dualstack-selection
bind-tcp [::]:5555  -no-speed-check -group foreign -no-dualstack-selection
server 202.102.128.68  -group domestic -exclude-default-group 
server-tls 223.5.5.5  -group domestic -exclude-default-group
server-tls 8.8.8.8  -group foreign -exclude-default-group
server-tls 1.1.1.1  -group foreign -exclude-default-group
server-tls 8.8.4.4  -group foreign -exclude-default-group
server-tls 1.12.12.12  
server 10.10.10.100:3053  -group domestic -exclude-default-group
server 10.10.10.100:1056  -group foreign -exclude-default-group
server 10.10.10.100:2053  
domain-set -name domain-forwarding-list -file /etc/smartdns/domain-forwarding.list
domain-rules /domain-set:domain-forwarding-list/ 
domain-set -name domain-block-list -file /etc/smartdns/domain-block.list
domain-rules /domain-set:domain-block-list/ --address #
conf-file /etc/smartdns/address.conf
conf-file /etc/smartdns/blacklist-ip.conf
conf-file /etc/smartdns/custom.conf



/etc/smartdns/custom.conf
bind :8055 -group domestic
bind-tcp :8055 -group domestic
serve-expired-ttl 259200000
serve-expired-prefetch-time 21600

**Reply by PikuZheng** [reply]
119是指`server-tls 1.12.12.12`这个吗？换成223.5.5.5就会快很多？而且只有ipv6结果非常慢，ipv4正常？正常情况下查询是ipv6和ipv4同时的，表现是同时慢吗

**Reply by yyysuo** [reply]
> 119是指`server-tls 1.12.12.12`这个吗？换成223.5.5.5就会快很多？而且只有ipv6结果非常慢，ipv4正常？正常情况下查询是ipv6和ipv4同时的，表现是同时慢吗

![image](https://user-images.githubusercontent.com/10081071/230805379-fe53a2fc-0b00-40ba-a3fe-76a8e2470c64.png)


dns架构用的**方案2**
不是，119指的是10.10.10.100:2053这个Adguard Home（第2个红框），Adguard Home里面只配置了一个119.29.29.29的情况下，ipv6的表现就超级慢，在第1个红框中的Adguard Home可以看到AAAA记录的查询时间都在200ms以上。换成223就没问题，一般是10ms以内。

**Reply by PikuZheng** [reply]
先考虑是不是119绕国外线路了。再就是运营商污染问题
